module/support/optimisation/NSGA2Evaluator/src/tasks/StandEvaluator.cpp

```diff
@@ -0,0 +1,160 @@
+                double sim_time = evaluator->sim_time;
+                sim_time++;
+
+                updateMaxFieldPlaneSway(sensors);
+                current_sensors = sensors;
+            }
+
+            void StandEvaluator::processOptimisationRobotPosition(const OptimisationRobotPosition& position) {
+                robot_position.x() = position.value.X;
+                robot_position.y() = position.value.Y;
+                robot_position.z() = position.value.Z;
+            }
+
+            void StandEvaluator::setUpTrial(const NSGA2EvaluationRequest& currentRequest) {
+                loadScript(currentRequest.task_config_path);
+                std::chrono::milliseconds limit_ms = std::chrono::milliseconds(0);
+                for (size_t i = 0; i < currentRequest.parameters.real_params.size(); i++) {
+                    int frame_time            = currentRequest.parameters.real_params[i];
+                    limit_ms                  = limit_ms + std::chrono::milliseconds(frame_time);
+                    script.frames[i].duration = std::chrono::milliseconds(frame_time);
+                }
+                saveScript(fmt::format("gen{:03d}_ind{:03d}_task-{}.yaml",
+                                       currentRequest.generation,
+                                       currentRequest.id,
+                                       currentRequest.task));
+
+                auto overhead =
+                    std::chrono::seconds(2);  // Overhead tacked on to give the robot time to fall over if unstable
+                trial_duration_limit = std::chrono::duration_cast<std::chrono::seconds>(limit_ms) + overhead;
+            }
+
+            void StandEvaluator::resetSimulation() {
+                // Reset our local state
+                trial_start_time     = 0.0;
+                robot_position       = Eigen::Vector3d::Zero();
+                max_field_plane_sway = 0.0;
+            }
+
+            void StandEvaluator::evaluatingState(size_t subsumptionId, NSGA2Evaluator* evaluator) {
+                NUClear::log<NUClear::DEBUG>("Running Script");
+                runScript(subsumptionId, evaluator);
+                NUClear::log<NUClear::DEBUG>("schedule expire");
+                evaluator->ScheduleTrialExpiredMessage(0, trial_duration_limit);
+            }
+
+            std::unique_ptr<NSGA2FitnessScores> StandEvaluator::calculateFitnessScores(bool earlyTermination,
+                                                                                       double sim_time,
+                                                                                       int generation,
+                                                                                       int individual) {
+                double trial_duration = sim_time - trial_start_time;
+                auto scores          = calculateScores(trial_duration);
+                auto constraints = earlyTermination ? calculateConstraints() : calculateConstraints();
```

Comment:
The original logic was that if the termination was not early, then none of the constraints were violated (on the assumption that the only way to get an early termination was to violate an optimisation constraint). It may be better to push all of that logic inside of `calculateConstraints` instead of here though, as it seems to have caused this odd line of code.